src/lib.rs

```diff
@@ -39,6 +39,9 @@ pub mod k3s;
 #[cfg(feature = "kafka")]
 #[cfg_attr(docsrs, doc(cfg(feature = "kafka")))]
 pub mod kafka;
+#[cfg(feature = "kafka_native")]
+#[cfg_attr(docsrs, doc(cfg(feature = "kafka_native")))]
+pub mod kafka_native;
```

Comment:
I'd suggest to keep it under `kafka` feature. We don't introduce any dev-time deps or something too specific.

WDYT of `kafka/native.rs`?

Reply:
It definitely makes sense to go under kafka, question is should the `confluent` image be phased out at some point? 
If it will be phased out, your suggestion would be non breaking change. 

I'd also suggest to name it `kafka/apache.rs`. 
Only thing missing is how would users be able to change between apache and confluent version, WDYT? any suggestions?

Reply:
I think we can put them into:
- `kafka/apache`
- `kafka/confluent`

And re-export module from `confluent` within `kafka` (with a comment about backward compatibility)
I think we may decide to keep both at the end and only breaking change in the future would be removal of re-exports.

Users can just import desired one.